clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -91,8 +122,22 @@ public NetworkClient(Selectable selector,
                          int socketReceiveBuffer,
                          int requestTimeoutMs,
                          Time time) {
-        this(null, metadata, selector, clientId, maxInFlightRequestsPerConnection,
-                reconnectBackoffMs, socketSendBuffer, socketReceiveBuffer, requestTimeoutMs, time);
+        this(NO_METADATA_UPDATER, metadata, selector, clientId, maxInFlightRequestsPerConnection,
+                reconnectBackoffMs, socketSendBuffer, socketReceiveBuffer, requestTimeoutMs, time, NO_APIS);
+    }
+
+    public NetworkClient(Selectable selector,
+                         MetadataUpdater metadataUpdater,
+                         String clientId,
+                         int maxInFlightRequestsPerConnection,
+                         long reconnectBackoffMs,
+                         int socketSendBuffer,
+                         int socketReceiveBuffer,
+                         int requestTimeoutMs,
+                         Time time,
+                         Collection<ApiVersionsResponse.ApiVersion> apis) {
```

Comment:
I don't think it has to be sorted, though having a consistent order will make it easier to read in logs. Also, I failed to see why changing this to a map is going to be more efficient. The checking is happening in `handleApiVersionsResponse` and response already has a map of keys to supported versions.